components/dealInDetails/sliderAndTimerComponents/Store.tsx

```diff
@@ -0,0 +1,223 @@
+import { Box, Typography, Divider, Link } from "@mui/material";
+import React, { useEffect, useState } from "react";
+import { all_center } from "../../../constant/commonStyle";
+import theme from "../../../theme";
+import CallIcon from "@mui/icons-material/Call";
+import LocalOfferOutlinedIcon from "@mui/icons-material/LocalOfferOutlined";
+import LanguageIcon from "@mui/icons-material/Language";
+import SendIcon from "@mui/icons-material/Send";
+import PlaceOutlinedIcon from "@mui/icons-material/PlaceOutlined";
+import { useSelector } from "react-redux";
+import StorefrontOutlinedIcon from "@mui/icons-material/StorefrontOutlined";
+import { useNavigate } from "react-router-dom";
+import Countdown from "react-countdown";
+import { Fullscreen } from "@mui/icons-material";
+
+const Store = () => {
+  const navigate = useNavigate();
+
+  const storeDescription = useSelector(
+    (store) => store.dealData.storeDescription
+  );
+
+  const [showAllText, setShowAllText] = useState(false);
+
+  const testString = storeDescription;
```

Comment:
use `useMemo` every where you are defined variables or computing value in components